base/linalg/dense.jl

```diff
@@ -29,6 +29,21 @@ function scale!(X::Array{T}, s::Real) where T<:BlasComplex
     X
 end
 
+function isone(x::StridedMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    for i in 1:m
+        for j in i:m
+            if i == j
+                @inbounds !isone(x[i,i]) && return false
+            else
+                @inbounds (!iszero(x[i,j]) || !iszero(x[j,i])) && return false
```

Comment:
Parenthesis around the `||` are not needed here.  But I was wondering: for big matrices which don't fit in cache, is it faster to do this triangular trick (loop over `m*n/2` indices), or to go through the matrice entries in a cache friendly way? I would believe the latter. You say this new version is a bit faster, but in the old one you were not looping in a cache-friendly, so I suggest you benchmark again this old version by swapping the `i` and `j` indices.